cmd/kubeadm/app/cmd/join.go

```diff
@@ -145,6 +145,11 @@ func (j *Join) Run(out io.Writer) error {
 		return err
 	}
 
+	err = kubenode.CheckForNodeNameDuplicates(connectionDetails)
```

Comment:
Yeah, I did at one point. Not sure what difference it makes if its before or after TLS bootstrap since we are passing the same object.